src/legacy/ui/public/vis/editors/default/components/sidebar/controls.tsx

```diff
@@ -0,0 +1,142 @@
+
+interface DefaultEditorControlsProps {
+  applyChanges(): void;
+  isDirty: boolean;
+  isInvalid: boolean;
+  isTouched: boolean;
+  dispatch: React.Dispatch<EditorAction>;
+  vis: Vis;
+}
+
+function DefaultEditorControls({
+  applyChanges,
+  isDirty,
+  isInvalid,
+  isTouched,
+  dispatch,
+  vis,
+}: DefaultEditorControlsProps) {
+  const { enableAutoApply } = vis.type.editorConfig;
+  const [autoApplyEnabled, setAutoApplyEnabled] = useState(false);
+  const toggleAutoApply = useCallback(e => setAutoApplyEnabled(e.target.checked), []);
+  const onClickDiscard = useCallback(() => dispatch(discardChanges(vis)), [dispatch, vis]);
+
+  useEffect(() => {
+    if (autoApplyEnabled && isDirty) {
+      applyChanges();
+    }
+  }, [isDirty, autoApplyEnabled]);
+
+  return (
+    <div className="visEditorSidebar__controls">
+      {!autoApplyEnabled && (
+        <EuiFlexGroup justifyContent="spaceBetween" gutterSize="none" responsive={false}>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="visualizeEditorResetButton"
+              disabled={!isDirty}
+              iconType="cross"
+              onClick={onClickDiscard}
+              size="s"
+            >
+              <FormattedMessage
+                id="common.ui.vis.editors.sidebar.discardChangesButtonLabel"
+                defaultMessage="Discard"
+              />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            {isInvalid && isTouched ? (
+              <EuiToolTip
+                content={i18n.translate('common.ui.vis.editors.sidebar.errorButtonTooltip', {
+                  defaultMessage: 'Errors in the highlighted fields need to be resolved.',
+                })}
+              >
+                <EuiButton color="danger" iconType="alert" size="s" disabled>
+                  <FormattedMessage
+                    id="common.ui.vis.editors.sidebar.updateChartButtonLabel"
+                    defaultMessage="Update"
+                  />
+                </EuiButton>
+              </EuiToolTip>
+            ) : (
+              <EuiButton
+                data-test-subj="visualizeEditorRenderButton"
+                disabled={!isDirty}
+                fill
+                iconType="play"
+                onClick={applyChanges}
+                size="s"
+              >
+                <FormattedMessage
+                  id="common.ui.vis.editors.sidebar.updateChartButtonLabel"
+                  defaultMessage="Update"
+                />
+              </EuiButton>
+            )}
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      )}
+      {enableAutoApply && (
+        <EuiToolTip
+          title={i18n.translate('common.ui.vis.editors.sidebar.autoApplyChangesLabel', {
+            defaultMessage: 'Auto apply is {autoApplyState}',
+            values: { autoApplyState: autoApplyEnabled ? 'on' : 'off' },
```

Comment:
I'd suggest using two separate messages for better  translation:
```ts
title={autoApplyEnabled
    ? i18n.translate('common.ui.vis.editors.sidebar.autoApplyChangesOnLabel', {
            defaultMessage: 'Auto apply is on',
      })
    : i18n.translate('common.ui.vis.editors.sidebar.autoApplyChangesOffLabel', {
            defaultMessage: 'Auto apply is off',
      })
}
```

Reply:
Done